src/tests/ftest/server/metadata.py

```diff
@@ -343,9 +387,12 @@ def test_metadata_addremove(self):
         :avocado: tags=ObjectMetadata,test_metadata_addremove
         """
         self.create_pool()
-        if not self.run_dummy_metadata_workload():
-            self.fail("failed to run dummy metadata workload")
-
+        # Skip dummy_metadata_workload when feature is disabled
+        svc_ops_enabled = self.pool.get_property("svc_ops_enabled")
+        if svc_ops_enabled:
+            svc_ops_entry_age = self.pool.get_property("svc_ops_entry_age")
+            if not self.run_dummy_metadata_workload(duration=svc_ops_entry_age):
+                self.fail("failed to run dummy metadata workload")
```

Comment:
We shouldn't conditionally run the test, and the default is enabled, so we should be able to just run this
```suggestion
        svc_ops_entry_age = self.pool.get_property("svc_ops_entry_age")
        if not self.run_dummy_metadata_workload(duration=svc_ops_entry_age):
            self.fail("failed to run dummy metadata workload")
```

Reply:
The suggestion does not work, since svc_ops_entry_age could be a valid number (ie: 60, or 300...) and svc_ops_enabled is disabled and causing issue.

Reply:
This calls `self.create_pool()`
Which has
```
def create_pool(self, svc_ops_enabled=True):
...
    if svc_ops_enabled:
        self.add_pool()   
```
So `svc_ops_enabled` is turned on

Reply:
no.  With the suggested 390-392, when svc_ops_enabled=False, the run_metsdata_dummy_workload will be executed.

Reply:
How would `svc_ops_enabled=False`? The default is True and this test doesn't pass `False`

Reply:
The testcase svc_ops_disabled will set svc_ops_enabled=False?

Reply:
This is `test_metadata_addremove`

Reply:
Yes. To determine running dummy_workload should check for svc_ops_enabled or not, not svc_ops_entry_age.

Reply:
`svc_ops_enabled` isn't dynamic here. It's always what is set in the config
https://github.com/daos-stack/daos/blob/05474cfc94b4018436bc648d6e3e442b2011d0f0/src/tests/ftest/server/metadata.yaml#L53

Reply:
For test_metadata_addremove(), for simplicity, I think we always want to run one variant (in this case, let's say we decide to use svc_ops_enabled:1 and run a dummy workload loop of svc_ops_entry_age seconds).

Which makes me wonder if in the .yaml file, maybe it should not specify svc_ops_enabled in the pool properties at all. And just leave it to the test cases in this code to specify in the argument to add_pool() whether svc_ops_enabled shall be True or False?

test_metadata_fillup_svc_ops_disabled and test_metadata_fillup_svc_ops_enabled already pass in the argument explicity.
test_metadata_addremove already effectively does the same (no argument means the default svc_ops_enabled=True is used).